x/programs/examples/imports/program/program.go

```diff
@@ -33,161 +33,161 @@ type Import struct {
 	engine  *engine.Engine
 	meter   *engine.Meter
 	imports host.SupportedImports
+	ctx     *program.Context
 }
 
 // New returns a new program invoke host module which can perform program to program calls.
-func New(log logging.Logger, engine *engine.Engine, mu state.Mutable, cfg *runtime.Config) *Import {
+func New(log logging.Logger, engine *engine.Engine, mu state.Mutable, cfg *runtime.Config, ctx *program.Context) *Import {
```

Comment:
nit: should we also take `program.Context` (not ptr) here? (overall seems switching to ptr would be better, your preference if you want to make a separate change)

Reply:
Going to leave this here and pass the pointer everywhere else (in a follow up)